### PR TITLE
refactor(jsonc): replace `enum` with string union type

### PR DESCRIPTION
**Pull request by realpha**
This PR replaces the used typescript enum in the `jsonc` module with a
`const object` as requested  [here](https://github.com/denoland/deno_std/issues/3782).

As this enum is only used internally, merging this change wouldn't constitute a
breaking change.

This change seems a little bit clunky but does the trick.

### CODE REVIEW

**Review by iuioiua — CHANGES_REQUESTED** [review]
Would changing `TokenType` to a union type make more sense in this case?

**Comment by realpha** [comment]
> Would changing `TokenType` to a union type make more sense in this case?

IMO yes. Wasn't sure if the convetion in the codebase in general is to prefer maps over simple string unions.

Will push the changes

File: jsonc/parse.ts [hunk 1]
```diff
@@ -51,30 +51,32 @@ export function parse(
...
+  EndObject: "EndObject",
+  BeginArray: "BeginArray",
+  EndArray: "EndArray",
+  NameSeparator: "NameSeparator",
+  ValueSeparator: "ValueSeparator",
+  NullOrTrueOrFalseOrNumber: "NullOrTrueOrFalseOrNumber",
+  String: "String",
+} as const;
+
+type TokenType = keyof typeof TokenTypeMap;
```
Comment (iuioiua) [inline jsonc/parse.ts:65]: I think it'd be simpler if we just had `TokenType` be a string union type without the intermediary `TokenTypeMap`. A quick skim suggested that might be fine.

**Review by iuioiua — CHANGES_REQUESTED** [review]

File: jsonc/parse.ts (hunk 1 above)
Comment (realpha) [reply jsonc/parse.ts]: Done

**Review by iuioiua — APPROVED** [review]
LGTM! Thank you.

**Pull request merged**